chalk-solve/src/clauses.rs

```diff
@@ -277,6 +293,36 @@ fn program_clauses_that_could_match<I: Interner>(
             let associated_ty_datum = db.associated_ty_data(alias.associated_ty_id);
             let trait_id = associated_ty_datum.trait_id;
             let trait_parameters = db.trait_parameters_from_projection(alias);
+
+            if (alias
+                .self_type_parameter(interner)
+                .bound(interner)
+                .is_some()
+                || alias
+                    .self_type_parameter(interner)
+                    .inference_var(interner)
+                    .is_some())
+                && (ty.bound(interner).is_some() || ty.inference_var(interner).is_some())
+            {
+                return Err(Floundered);
+            }
+
+            let trait_datum = db.trait_datum(trait_id);
+
+            // FIXME
```

Comment:
Why FIXME?

Reply:
// Flounder if the self-type is unknown and the trait is non-enumerable.
//
// e.g., `Normalize(<?X as Iterator>::Item = u32)`

Reply:
Just FIXME because I went back and forth a few times on this and wasn't sure whether to keep it :sweat_smile: